packages/core/src/agent/index.ts

```diff
@@ -516,23 +516,15 @@ export class Agent<
           content: `${this.instructions}. Today's date is ${new Date().toISOString()}`,
         };
 
-        const userMessages: CoreMessage[] = messages.map(content => ({
-          role: 'user',
-          content: content,
-        }));
```

Comment:
since we previously converted all core messages into an array of strings, this code would then make every message have `role: "user"`